src/compile/data.ts

```diff
@@ -667,16 +690,63 @@ export namespace summary {
     return [];
   }
 
+  function mergeMeasures(parentMeasures, childMeasures) {
+    for (var field in childMeasures) {
```

Comment:
Oh yeah. I saw this comment and wanted so say YES! 
Maybe use this https://github.com/basarat/typescript-collections